c_src/erl_optics.c

```diff
@@ -511,13 +548,17 @@ static ERL_NIF_TERM eo_optics_poll(
     if (!optics) return ERROR("get_optics");
     if (!poller) return ERROR("poller_allocation_error");
 
-    ctx->env = env;
-    ctx->map = enif_make_new_map(env);
-
+    if (erlang_poller != 0){
+        ctx->env = env;
```

Comment:
Do you know what the lifetime of the `env` object is? Is it safe to use it after returning from this function?